Source/MMWormholeManifestFileTransiting.m

```diff
@@ -0,0 +1,140 @@
+//
+// MMWormholeCoordinatedFileTransiting.h
+//
+// Copyright (c) 2015 Mutual Mobile (http://www.mutualmobile.com/)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MMWormholeManifestFileTransiting.h"
+
+@implementation MMWormholeManifestFileTransiting
+
+#pragma mark - MMWormholeTransiting Methods
+
+- (BOOL)writeMessageObject:(id<NSCoding>)messageObject forIdentifier:(NSString *)identifier {
+    if (identifier == nil) {
+        return NO;
+    }
+    
+    if (messageObject) {
+        //Insert into manifest
+        NSMutableArray *storageArray = nil;
+        NSArray *potentialOnDiskArray = (NSArray *)[self _arrayForIdentifier:identifier];
+        if (potentialOnDiskArray) {
+            storageArray = [NSMutableArray arrayWithArray:potentialOnDiskArray];
+        } else {
+            storageArray = [[NSMutableArray alloc] init];
+        }
+        [storageArray addObject:messageObject];
+        
+        //Encode manifest instead of message object
+        BOOL success = [self _writeArrayToDisk:storageArray forIdentifier:identifier];
+        
+        if (!success) {
+            return NO;
+        }
+    }
+    
+    return YES;
+}
+
+- (id<NSCoding>)messageObjectForIdentifier:(NSString *)identifier {
+    if (identifier == nil) {
+        return nil;
+    }
+    
+    NSArray *storageArray = [self _arrayForIdentifier:identifier];
+    if ([storageArray count] == 0) {
+        return nil;
+    }
+    id messageObject = storageArray[0];
```

Comment:
Seems curious that we are making the assumption the array returned for this `identifier` is always a single item collection. Why are we writing and reading arrays from the file system and not the message object itself?